pkg/view/component/dropdown.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright (c) 2020 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package component
+
+import (
+	"encoding/json"
+)
+
+// DropdownType defines the dropdown type
+// basically what component is used as dropdown source
+// visible when dropdown is closed
+type DropdownType string
+
+const (
+	DropdownButton DropdownType = "button"
+	DropdownLink   DropdownType = "link"
+	DropdownLabel  DropdownType = "label"
+	DropdownIcon   DropdownType = "icon"
+)
+
+// DropdownPosition denotes a position relative to source button
+type DropdownPosition string
+
+const (
+	BottomLeft  DropdownPosition = "bottom-left"
+	BottomRight DropdownPosition = "bottom-right"
+	TopLeft     DropdownPosition = "top-left"
+	TopRight    DropdownPosition = "top-right"
+	LeftBottom  DropdownPosition = "left-bottom"
+	LeftTop     DropdownPosition = "left-top"
+	RightTop    DropdownPosition = "right-top"
+	RightBottom DropdownPosition = "right-bottom"
+)
+
+// Defines the type of dropdown item
+type ItemType string
+
+const (
+	Header    ItemType = "header"
+	PlainText ItemType = "text"
+	Url       ItemType = "link"
+	Separator ItemType = "separator"
+)
+
+type DropdownItemConfig struct {
+	Name        string   `json:"name"`
+	Type        ItemType `json:"type"`
+	Label       string   `json:"label"`
+	Url         string   `json:"url,omitempty"`
+	Description string   `json:"description,omitempty"`
+}
+
+// DropdownConfig defines the contents of a Dropdown
+type DropdownConfig struct {
+	DropdownPosition DropdownPosition     `json:"position,omitempty"`
+	DropdownType     DropdownType         `json:"type"`
+	Action           string               `json:"action,omitempty"`
+	UseSelection     bool                 `json:"useSelection"`
+	Items            []DropdownItemConfig `json:"items"`
+}
+
+// Dropdown: displays dropdown component with a list of values
+// Used to choose an option or action from a contextual list
+// +octant:component
+type Dropdown struct {
+	Base
+	Config DropdownConfig `json:"config"`
+}
+
+// NewDropdown creates a new dropdown component
+func NewDropdown(title string, position DropdownPosition, dropdownType DropdownType, action string, useSelection bool, items ...DropdownItemConfig) *Dropdown {
+	dropdownItems := append([]DropdownItemConfig(nil), items...)
+	return &Dropdown{
+		Base: newBase(TypeDropdown, TitleFromString(title)),
+		Config: DropdownConfig{
+			DropdownPosition: position,
+			DropdownType:     dropdownType,
+			Action:           action,
+			UseSelection:     useSelection,
+			Items:            dropdownItems,
+		},
+	}
+}
+
+// AddDropdownItem adds an item to the dropdown
+func (t *Dropdown) AddDropdownItem(name string, itemType ItemType, label string, url string, description string) {
+	item := DropdownItemConfig{
```

Comment:
Name is used to identify the item (look the action payload) and Description is not currently used but it may in future for tooltip or something.  Everything should have Description field IMO :-)